### PR TITLE
update Tendermint genesis

### LINKED ISSUE #916: Tendermint genesis is outdated
The genesis module doesn't seem to be compatible with Tendermint v0.34.x:
- missing `initial_height` field
- missing `consensus.block.time_iota_ms`
- incompatible type for `app_hash`
- `validators.voting_power` is not accepted by Tendermint Go, the default should be `validators.power`
- missing optional field `validators.name`

### PR DESCRIPTION
**Pull request by tzemanovic**
closes #916 

* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Added entry in `.changelog/`

### CODE REVIEW

File: tendermint/src/genesis.rs [hunk 1]
```diff
@@ -24,25 +25,9 @@ pub struct Genesis<AppState = serde_json::Value> {
     pub validators: Vec<validator::Info>,
 
     /// App hash
-    #[serde(skip_serializing_if = "Vec::is_empty", with = "serde_bytes")]
-    pub app_hash: Vec<u8>,
+    pub app_hash: String,
```
Comment (thanethomson) [inline tendermint/src/genesis.rs:28]: What's your reasoning behind this change?

**Review by thanethomson — COMMENTED** [review]
Just one question from my side. Otherwise looks good! :pray:

File: tendermint/src/genesis.rs (hunk 1 above)
Comment (tzemanovic) [reply tendermint/src/genesis.rs:28]: In our app, we're updating the genesis.json generated from Tendermint Go. It sets `"app_hash": ""`, which is parsed as an empty `Vec`, but when we serialize it back to json, it omits the field completely. Then on subsequent run, the parser fails due to `Error("missing field `app_hash`", ...)`

Comment (thanethomson) [reply tendermint/src/genesis.rs:28]: And what if you changed it to:

```suggestion
    #[serde(with = "serde_bytes")]
    pub app_hash: Vec<u8>,
```

In the Go version of Tendermint it's modeled as `HexBytes`, which is an array of bytes, and I think it should be modeled similarly here.

Comment (tzemanovic) [reply tendermint/src/genesis.rs:28]: unfortunately, this writes back `"app_hash": []` into JSON, which is not accepted :/

Comment (tzemanovic) [reply tendermint/src/genesis.rs:28]: perhaps this could use a custom serialize/deserialize with hex encoding/decoding

Comment (tzemanovic) [reply tendermint/src/genesis.rs:28]: `#[serde(with="serializers::bytes::hexstring")]` does it, I'll fix it up

**Review by thanethomson — APPROVED** [review]
LGTM! :+1:

**Pull request merged**